components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2762,7 +2766,7 @@ private int addAuthenticator(Connection conn, int tenantId, String idpName,
             }
         } finally {
             IdentityDatabaseUtil.closeResultSet(resultSet);
-            IdentityDatabaseUtil.closeStatement(roadPermissionsPrepStmt);
+            IdentityDatabaseUtil.closeStatement(readPermissionsPrepStmt);
```

Comment:
Now that we are using a different connection for UM DB we must close it. Earlier we didn't close the connection that was passed because we were using it to do a transaction.